packages/admin/admin/src/error/errorboundary/ErrorBoundary.tsx

```diff
@@ -17,118 +17,213 @@ export type ErrorBoundaryClassKey =
+        padding-left: ${theme.spacing(1)};
+    `,
+);
+
+const ExceptionStackTrace = styled(Typography, {
+    name: "CometAdminErrorBoundary",
+    slot: "exceptionStackTrace",
+    overridesResolver(_, styles) {
+        return [styles.exceptionStackTrace];
+    },
+})();
+
+export const ErrorBoundary = (inProps: ErrorBoundaryProps) => {
+    const props = useThemeProps({ props: inProps, name: "CometAdminErrorBoundary" });
+    return <CoreErrorBoundary {...props} />;
+};
+
+class CoreErrorBoundary extends React.Component<ErrorBoundaryProps, IErrorBoundaryState> {
+    constructor(props: ErrorBoundaryProps) {
         super(props);
         this.state = {};
     }
 
     public componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
-        this.setState({ error, errorInfo });
+        this.setState((prev) => ({ ...prev, error, errorInfo }));
     }
 
     public render() {
         const {
-            classes,
             variant = "filled",
             icon = <Error />,
             toggleDetailsOpenedIcon = <ChevronRight fontSize="small" />,
             toggleDetailsClosedIcon = <ChevronDown fontSize="small" />,
+            children,
+            slotProps,
+            userErrorMessage,
+            key,
+            ...restProps
         } = this.props;
-        const { error, errorInfo } = this.state;
+        const { error, errorInfo, showDetails } = this.state;
+
+        const ownerState: OwnerState = {
+            showDetails,
+        };
 
         if (errorInfo != null) {
             return (
-                <Alert variant={variant} icon={icon} severity="error" classes={{ root: classes.alert }}>
-                    <Typography classes={{ root: classes.message }}>
-                        {this.props.userErrorMessage ? (
-                            this.props.userErrorMessage
+                <Alert variant={variant} icon={icon} severity="error" {...restProps} {...slotProps?.alert}>
+                    <Message {...slotProps?.message}>
+                        {userErrorMessage ? (
+                            userErrorMessage
                         ) : (
                             <FormattedMessage id="comet.error.abstractErrorMessage" defaultMessage="An error has occurred" />
                         )}
-                    </Typography>
+                    </Message>
 
                     {process.env.NODE_ENV === "development" && (
-                        <details className={classes.exceptionDetails}>
-                            <summary className={classes.exceptionSummary}>
-                                <div className={`${classes.exceptionSummaryIcon} ${classes.exceptionSummaryIconOpened}`}>
+                        <ExceptionDetails
+                            onToggle={(e) => {
+                                const showDetails = e.currentTarget.hasAttribute("open");
+                                this.setState((prev) => ({ ...prev, showDetails }));
+                            }}
```

Comment:
This is a new functionality, right? If yes, why did you add it? Could you please provide a screencast?

Reply:
This doesn't change the behavior. 

Previously the icons where shown/hidden in the styling by selecting `details[open] .{icon-class}`. 
Like this:

```ts
exceptionDetails: {
    "&[open]": {
        "& $exceptionSummaryIconClosed": {
            display: "flex",
        },
        "& $exceptionSummaryIconOpened": {
            display: "none",
        },
    },
},
```

This is no longer possible, because the Details and the Icon are separate slots which cannot reference another slot, like you could do in styled-components for example:

```ts
const IconSlot = styled(...);
const DetailsSlot = styled(...)(css`
  &[open] ${IconSlot} { // Selecting `IconSlot` is not supported by MUIs `styled()`
    // show or hide
  }
`)
```

That's why I added an additional `open` value to the state, that just mirrors the `open` attribute on the `Details` element which changes on `onToggle`.